### DISCUSSION TITLE
Don't apply syntax highlighting to LICENSE files

### DISCUSSION THREAD

**Discussion by jwortmann** [github-linguist/linguist#6493] [Classification]
GitHub should not try to apply syntax highlighting to files named "LICENSE" (without file extension).

Example: https://github.com/apache/flink/blob/7b5fe279cdebaa7b712141da286cde2f864c6bbf/LICENSE

<details><summary>Screenshot</summary>

![apache-license](https://github.com/github-linguist/linguist/assets/6579999/29f84921-31a7-49cf-b7d2-71fe11699395)

</details>

According to

https://github.com/github-linguist/linguist/blob/e855ef2b6f90c34074061a2e17acbe853e61b483/lib/linguist/languages.yml#L7010

it seems like it's already the intended behavior to highlight such files as plain text (and I'm quite sure it did work in the past), but apparently the language is detected incorrectly at the moment.

**Comment by lildude** [comment]
Yes, this shouldn't be happening as the `tm_scope` is not set which is how Linguist influences the syntax highlighting:

https://github.com/github-linguist/linguist/blob/e855ef2b6f90c34074061a2e17acbe853e61b483/lib/linguist/languages.yml#L7027

I've just cloned the repo and analysed the file directly and it correctly identifies it as text so something else has changed in the rendering of files on GitHub independent of Linguist or there's something specific to this repo. Interestingly, this doesn't apply to all LICENSE files... the LICENSE file in this repo isn't highlighted and neither is the Apache license in other repos.

Please use the "Contact GitHub" link on the bottom of the page showing the problem to log a support ticket for which can be routed to the correct team as something else is at play and I'm not sure what it is. Feel free to reference this discussion.

**Reply by jwortmann** [reply]
Thanks for the answer.

> and neither is the Apache license in other repos

It is not exclusive to the file in the linked repo though; I can see it happen also for some other Apache license files, for example in a repo of mine: https://github.com/jwortmann/ini-syntax/blob/master/LICENSE (this is where I saw it initially). But you are right that it doesn't happen for all Apache license files.
